interfaces/tempo_cluster/v0/schema.py

```diff
@@ -5,30 +5,36 @@
 - RequirerSchema
 """
 from enum import Enum
-from typing import Optional, Dict, Any, Literal
+from typing import Optional, Dict, Any, Literal, MutableMapping, List
+
+import json
 
 from interface_tester.schema_base import DataBagSchema
-from pydantic import BaseModel, Field
+from pydantic import BaseModel, ConfigDict, Field, ValidationError, Json
 
 
 class TempoClusterProviderAppData(BaseModel):
     """TempoClusterProviderAppData."""
-    tempo_config: str = Field(
+    worker_config: Json[str] = Field(
```

Comment:
The description says that this is YAML. Are you using `Json` in the "subset of YAML" meaning here? Or should the description be changed?

Reply:
This is a convention that we use across all our interfaces: we are json-encoding the contents of the databags. You can see a sample implementation [in the shared worker-coordinator object](https://github.com/canonical/cos-lib/blob/main/src/cosl/coordinated_workers/interface.py#L58).

You can see that ingress is following the same approach [here](https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/ingress/v2/schema.py),  same with [does prometheus_scrape too](https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/prometheus_scrape/v0/schema.py#L128).

Reply:
The field itself is parsed as string, then converted into yaml. So `Json` is treated as a wrapper here.